components/artist_page_cta/view.coffee

```diff
@@ -26,7 +26,9 @@ module.exports = class ArtistPageCTAView extends Backbone.View
     @alreadyDismissed = false
     @afterAuthPath = "#{@artist.get('href')}/payoff"
     @$window.on 'scroll', _.throttle(@maybeShowOverlay, 200)
-    
+    mediator.on 'clickFollowButton', @fullScreenOverlay
```

Comment:
Use mediator to toggle the overlay when follow button, or the header auth (login/signup) is clicked.